spec/controllers/api/v0/order_cycles_controller_spec.rb

```diff
@@ -70,6 +70,26 @@ module Api
           expect(product_ids).to include product1.id, product2.id
           expect(product_ids).to_not include product3.id
         end
+
+        context "with supplier properties" do
+          let!(:supplier_property) { create(:property, presentation: 'Certified Organic') }
+          let!(:supplier) { create(:supplier_enterprise, properties: [supplier_property]) }
+
+          before do
+            product1.update!(supplier: supplier)
+            product2.update!(supplier: supplier)
+            product3.update!(supplier: supplier)
+            product4.update!(supplier: supplier, inherits_properties: false)
+          end
+
+          it "filter out the product that don't inherits from supplier properties" do
+            api_get :products, id: order_cycle.id, distributor: distributor.id,
+                               q: { properties_id_or_supplier_properties_id_in_any: [supplier_property.id] }
+
+            expect(product_ids).to include product1.id, product2.id, product3.id
+            expect(product_ids).to_not include product4.id
```

Comment:
If you look at the order cycle setup then product4 is not included in the OC, regardless of any filters, it won't be in the list. So maybe ignore product4 and just work with product1, 2 and 3.

Reply:
Right. Will add it.